tools/actions_local_runner.py

```diff
@@ -70,65 +78,169 @@ def find_changed_files() -> List[str]:
     return [x.strip() for x in all_files if x.strip() != ""]
 
 
-async def run_step(step: Dict[str, Any], job_name: str, files: Optional[List[str]], quiet: bool) -> bool:
+def print_results(job_name: str, passed: bool, streams: List[str]) -> None:
+    header(job_name, passed)
+    for stream in streams:
+        stream = stream.strip()
+        if stream != "":
+            print(stream)
+
+
+async def shell_cmd(
+    cmd: str, env: Optional[Dict[str, Any]] = None, redirect: bool = True
+) -> Tuple[bool, str, str]:
+    proc = await asyncio.create_subprocess_shell(
+        cmd,
+        shell=True,
+        cwd=REPO_ROOT,
+        env=env,
+        stdout=subprocess.PIPE if redirect else None,
+        stderr=subprocess.PIPE if redirect else None,
+        executable=shutil.which("bash"),
+    )
+    stdout, stderr = await proc.communicate()
+
+    passed = proc.returncode == 0
+    if not redirect:
+        return passed, "", ""
+
+    return passed, stdout.decode(), stderr.decode()
+
+
+def header(name: str, passed: bool) -> None:
+    PASS = color(col.GREEN, "\N{check mark}")
+    FAIL = color(col.RED, "x")
+    icon = PASS if passed else FAIL
+    print(f"{icon} {color(col.BLUE, name)}")
+
+
+def get_flake_excludes() -> List[str]:
+    config = configparser.ConfigParser()
+    config.read(os.path.join(REPO_ROOT, ".flake8"))
+
+    excludes = config["flake8"]["exclude"].split("\n")
+    return [i.strip() for i in excludes if i.strip() != ""]
```

Comment:
this seems incorrect to me; when I run this locally, I get the following:
```py
['./.git,',
 './build_code_analyzer,',
 './build_test_custom_build,',
 './build,',
 './caffe2,',
 './docs/caffe2,',
 './docs/cpp/src,',
 './docs/src,',
 './scripts,',
 './test/generated_type_hints_smoketest.py,',
 './third_party,',
 './torch/include,',
 './torch/lib,',
 './venv,',
 '*.pyi']
```
I'm guessing the commas are not supposed to be there? in any case, maybe you could use the technique from `tools/mypy_wrapper.py` instead:

https://github.com/pytorch/pytorch/blob/b5b158a6c6de94dfb983b447fa33fea062358844/tools/mypy_wrapper.py#L61

```suggestion
    return re.split(r',\s*', config["flake8"]["exclude"].strip())
```

Reply:
I guess you could also extract this out into a utility function shared by both `tools/mypy_wrapper.py` and `tools/actions_local_runner.py`, if you want